sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -67,10 +98,12 @@ private static HttpPipeline buildPipeline(TokenCredential tokenCredential, Strin
 
         policies.addAll(additionalPolicies);
 
-        // Custom policies, authentication policy, and add date policy all take place after the retry policy
+        // Custom policies, authentication policy, and add date policy all take place after the retry policy which means
+        // they will be applied once per retry. For instance, the AddDatePolicy will add a different date time header
+        // each time the retry is attempted.
```

Comment:
Q - do we know if `AddDatePolicy` updates the date per retry attempt, or it updates the date for each http request made?

Reply:
Since that policy is applied after the retry policy, each retry message will have a date time header that fits when the retry attempt occurred

Reply:
I am not sure if I understand this correctly:

1. A client is created, date policy is set.
2. Make http request (1) with this client -> current date is added to the request - request is a success.
3. Make http request (2) with this client -> current date is added to the request - request fails -> retry kicks in -> new http request (3) is made.

Since the current date is sent with request (1), (2) and (3), how does retry matter here?
Or am I missing something?

For auth, I can understand, that the next time retrt happens, you also want the auth credentials to be refreshed; but I cannot understand why date policy is relevant with retry.

Reply:
We discussed this offline, and I'm putting up another PR to address my confusing wording of this #14283